### PR TITLE
Implement failed snapshot pop-up on task screen [SCI-6612]

### PR DESCRIPTION
**Pull request by artoscinote**
Jira ticket: [SCI-6612](https://biosistemika.atlassian.net/browse/SCI-6612)

### What was done
Added general purpose 'dialog' modal.
Implement failed snapshot pop-up on task screen.

### CODE REVIEW

File: app/views/shared/_dialog.html.erb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      </div>
+    </div>
+  </div>
+</div>
+
+<% if defined?(shown) && shown %>
+  <script>
+    $('#<%= id %>').modal('show');
+  </script>
+<% end %>
```
Comment (aignatov-bio) [inline app/views/shared/_dialog.html.erb:38]: If this shared element just move this logic to sitewide

File: app/views/shared/_dialog.html.erb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          <% when 'info' %>
+            <i class="fas fa-info"></i>
+          <% when 'success' %>
+            <i class="fas fa-check-circle"></i>
+          <% else %>
+            <i class="fas fa-info"></i>
+          <% end %>
+        </div>
+        <div class="sci-dialog__body">
+          <%= body.html_safe %>
```
Comment (aignatov-bio) [inline app/views/shared/_dialog.html.erb:22]: Why html_safe?

Comment (artoscinote) [reply app/views/shared/_dialog.html.erb:22]: Because in this case, the translation includes some bold tags.

Comment (aignatov-bio) [reply app/views/shared/_dialog.html.erb:22]: Hm does `_html` suffix in I18n fix this?

Comment (artoscinote) [reply app/views/shared/_dialog.html.erb]: Ah, didn't know that, yes it works without .html_safe. Fix pushed.

**Pull request merged**